x-pack/plugins/reporting/server/plugin.ts

```diff
@@ -49,22 +43,25 @@ export class ReportingPlugin
     });
 
     const { elasticsearch, http } = core;
-    const { licensing, security } = plugins;
+    const { licensing, security, taskManager } = plugins;
     const { initializerContext: initContext, reportingCore } = this;
 
     const router = http.createRouter();
     const basePath = http.basePath.get;
+    const task = new ReportingTask(this.reportingCore, this.logger);
 
     reportingCore.pluginSetup({
       elasticsearch,
       licensing,
       basePath,
       router,
       security,
+      task,
     });
 
     registerReportingUsageCollector(reportingCore, plugins);
     registerRoutes(reportingCore, this.logger);
+    task.setup(taskManager);
```

Comment:
`task.setup` should have a `config` argument passed, so that registering the task can use the `timeout` and `maxAttempts` setting from config. 

Unfortunately, `task.setup` can not be moved down into the async background setup function: TM ignores non-synchronous task type registration. 

To solve this, we need: https://github.com/elastic/kibana/issues/74179